templates/msg.h.em

```diff
@@ -176,8 +176,14 @@ bool _@(msg_underscored_name)_decode(const CanardRxTransfer* transfer, uint32_t*
 @[  if msg_union]@
 @(ind)@(union_msg_tag_uint_type_from_num_fields(len(msg_fields))) union_tag;
 @(ind)canardDecodeScalar(transfer, *bit_ofs, @(union_msg_tag_bitlen_from_num_fields(len(msg_fields))), false, &union_tag);
-@(ind)msg->union_tag = (enum @(msg_underscored_name)_type_t)union_tag;
 @(ind)*bit_ofs += @(union_msg_tag_bitlen_from_num_fields(len(msg_fields)));
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wtype-limits"
+@(ind)if (union_tag > @(len(msg_fields)-1)) {
```

Comment:
could len(msg_fields) be zero here? should be check for union_tag >= len(msg_fields) instead?

Reply:
No, it's not possible: https://github.com/dronecan/pydronecan/blob/8dd60c5abebde41260eccebb625f2d22449af11f/dronecan/dsdl/parser.py#L804

Reply:
the >= check is still preferable I think